src/chain_sync/chain_muxer.rs

```diff
@@ -850,6 +901,22 @@ where
                         self.state = ChainMuxerState::Connect(self.evaluate_network_head());
                     }
                 }
+                ChainMuxerState::Stateless(ref mut either) => match either {
+                    Either::Left(ref mut connect) => {
+                        if let Err(e) = std::task::ready!(connect.as_mut().poll(cx)) {
+                            error!("Failed to process network messages, retrying. Error = {e:?}");
+                        }
+                        self.state = ChainMuxerState::Stateless(Either::Right(Box::pin(
+                            tokio::time::sleep(Duration::from_secs(60)),
+                        )));
+                    }
+                    Either::Right(ref mut sleep) => {
+                        std::task::ready!(sleep.as_mut().poll(cx));
+                        self.state = ChainMuxerState::Stateless(Either::Left(
+                            self.handle_network_messages(),
+                        ));
+                    }
```

Comment:
`handle_network_messages` loops forever. So there's no need for `sleep`. Or is there something I'm missing?

Reply:
Yes, the loop breaks only when `p2p_messages` channel is dropped, changed to returning `Poll::Ready` in that case